semgrep-core/src/analyzing/Naming_AST.ml

```diff
@@ -587,7 +597,13 @@ let resolve2 lang prog =
             add_ident_current_scope id resolved env.names;
             set_resolved env id_info resolved;
             k x
-        | OtherPat _ ->
+        | OtherPat _
+          (* This interacts badly with implicit JS/TS declarations. It causes
```

Comment:
add a big TODO here, that says we should fix the AST of JS/TS so those f({ foo}) patterns do not show
as regular variables.
I think @mjambon wanted to rewrite the pattern part of ast_js.ml at some point.

Reply:
Done! FYI yesterday I opened a related issue: #3103.